bindings/matrix-sdk-ffi/src/room.rs

```diff
@@ -233,31 +233,36 @@ impl Room {
         &self,
         listener: Box<dyn TimelineListener>,
     ) -> RoomTimelineListenerResult {
-        let timeline = self
-            .timeline
-            .write()
-            .await
-            .get_or_insert_with(|| {
-                let room = self.inner.clone();
-                let timeline = RUNTIME.block_on(room.timeline());
-                Arc::new(timeline)
-            })
-            .clone();
+        let timeline = self.timeline.clone();
+        let room = self.inner.clone();
+        RUNTIME
+            .spawn(async move {
+                let timeline = timeline
+                    .write()
+                    .await
+                    .get_or_insert_with(|| {
+                        let timeline = RUNTIME.block_on(room.timeline());
+                        Arc::new(timeline)
+                    })
+                    .clone();
 
-        let (timeline_items, timeline_stream) = timeline.subscribe().await;
+                let (timeline_items, timeline_stream) = timeline.subscribe().await;
 
-        let timeline_stream = TaskHandle::new(RUNTIME.spawn(async move {
-            pin_mut!(timeline_stream);
+                let timeline_stream = TaskHandle::new(RUNTIME.spawn(async move {
+                    pin_mut!(timeline_stream);
 
-            while let Some(diff) = timeline_stream.next().await {
-                listener.on_update(Arc::new(TimelineDiff::new(diff)));
-            }
-        }));
+                    while let Some(diff) = timeline_stream.next().await {
+                        listener.on_update(Arc::new(TimelineDiff::new(diff)));
+                    }
+                }));
 
-        RoomTimelineListenerResult {
-            items: timeline_items.into_iter().map(TimelineItem::from_arc).collect(),
-            items_stream: Arc::new(timeline_stream),
-        }
+                RoomTimelineListenerResult {
+                    items: timeline_items.into_iter().map(TimelineItem::from_arc).collect(),
+                    items_stream: Arc::new(timeline_stream),
+                }
+            })
+            .await
+            .unwrap()
```

Comment:
Can we get rid of these unwraps, perhaps create a static string that is shared between those and used in each `expect()`.

Reply:
Hmm, I suppose that's correct.